src/clarity-angular/select/select.ts

```diff
@@ -0,0 +1,185 @@
+
+    ngOnDestroy() {
+        this._subOpen.unsubscribe();
+        this._subInput.unsubscribe();
+        this._subSpecialKeys.unsubscribe();
+    }
+    /**
+     * Opens the select menu on input and filters the options
+     *
+     * @param {string} input
+     * @memberof Select
+     */
+    handleInput(input: string) {
+        const state: State = {search: input};
+        this.refresh.emit(state);
+        if (input !== "") {
+            this.ifOpenService.open = true;
+            this.options.map((option) => {
+                // this is quite restrictive as it its case-sensitive right now, is that expected behaviour ?
+                if (input && option.clrTitle.indexOf(input) === -1) {
+                    option.visible = false;
+                } else {
+                    option.visible = true;
+                }
+            });
+        } else {
+            this.ifOpenService.open = false;
+            this.options.map((option) => {
+                option.visible = true;
+            });
+        }
+        if (this.selectService.selected &&
+            input.toLowerCase().indexOf(this.selectService.selected.clrTitle.toLowerCase()) === -1) {
+            this.selectService.selected = null;
+            this.ngModel = null;
+        }
+    }
+    /**
+     * Handles ArrowUp, -Down and Enter
+     *
+     * @param {string} key
+     * @returns
+     * @memberof Select
+     */
+    handleSpecialKeys(key: string) {
+        if (key === "ArrowUp") {
+            this.selectService.highlighted = this.getPrevOption(this.selectService.highlighted);
+        } else if (key === "ArrowDown") {
+            if (!this.ifOpenService.open) {
+                this.ifOpenService.open = true;
+                const state: State = {search: ""};
+                this.refresh.emit(state);
+                return;
+            }
+            this.selectService.highlighted = this.getNextOption(this.selectService.highlighted);
+        } else if (key === "Enter") {
+            if (this.ifOpenService.open && this.selectService.highlighted != null) {
+                this.selectOption(this.selectService.highlighted);
+                this.ifOpenService.open = false;
+            }
+        }
+    }
+    selectOption(option: Option) {
+        this.selectService.selected = option;
+        this.ngModel = option.clrValue;
+        this.selectService.input = option.clrTitle;
+    }
+    getPrevOption(pivotOption: Option): Option {
+        const availableOptions = this.options.filter((option: Option) => {
+            return option.visible;
+        });
+        if (availableOptions[0] === pivotOption) {
+            return pivotOption;
+        }
+        const elementPos = availableOptions
+                               .map((x) => {
+                                   return x.clrValue;
+                               })
+                               .indexOf(pivotOption.clrValue);
```

Comment:
This breaks when several options have the same value (which is a valid use case).